item-impl/src/main/java/com/example/auction/item/impl/PItemEntity.java

```diff
@@ -50,6 +53,14 @@ private Behavior created(PItemState state) {
 
         builder.setReadOnlyCommandHandler(GetItem.class, this::getItem);
 
+        builder.setCommandHandler(UpdateItem.class, (updateItem, ctx) -> {
+                    ItemUpdated event = ItemUpdated.from(updateItem);
```

Comment:
I was (wrongly) asserting on the service that the principal of the update action was equal to the creatorId in the payload of the request. I will change it so that the principal is considered the _commander_ on the  `UpdateItem`. Then the PersistentEntity can assert that the `creator` matches the `commander`.
By not using `Pitem` as the payload of `UpdateItem` many of the checks you suggest are not necessary.

I did forget to block the event generation when no field changes.